libs/zglfw/src/zglfw.zig

```diff
@@ -233,6 +244,147 @@ pub const CursorMode = enum(i32) {
 };
 //--------------------------------------------------------------------------------------------------
 //
+// Joystick
+//
+//--------------------------------------------------------------------------------------------------
+
+pub const Joystick = struct {
+    pub const ButtonAction = enum(u8) {
+        release = 0,
+        press = 1,
+    };
+
+    jid: u4,
+
+    pub const maximum_supported = 16;
+
+    pub fn init(jid: u4) ?Joystick {
+        const self: Joystick = .{ .jid = jid };
+        if (!self.present()) {
+            return null;
+        }
+        return self;
+    }
+
+    pub fn present(self: Joystick) bool {
+        return glfwJoystickPresent(@intCast(i32, self.jid)) == @boolToInt(true);
+    }
+    extern fn glfwJoystickPresent(jid: i32) i32;
+
+    pub fn getGUID(self: Joystick) [*:0]const u8 {
```

Comment:
Let's call it `getGuid()` (Zig convention). Return type should be `[:0]` (requires `std.mem.span()` in impl).